src/foam/nanos/auth/UserAndGroupAuthService.java

```diff
@@ -141,21 +147,18 @@ public User login(X x, long userId, String password) throws AuthenticationExcept
       throw new AuthenticationException("Invalid Parameters");
     }
 
-    User user = (User) userDAO_.find(userId);
+    User user = (User) userSubclassDAO_.find(userId);
     User contextUser = (User) userAndGroupContext(x, user, password);
     return contextUser;
   }
 
   public User loginByEmail(X x, String email, String password) throws AuthenticationException {
-    Sink sink = new ArraySink();
-    sink = userDAO_.where(MLang.EQ(User.EMAIL, email.toLowerCase())).limit(1).select(sink);
+    User user = (User) userDAO_.find(MLang.EQ(User.EMAIL, email.toLowerCase()));
```

Comment:
Why is this also not `userSubclassDAO_`?

Reply:
Because subclasses of User can have the same email address as a User, but we don't want to find the subclass, we want to find the User.